### PR TITLE
Add `typos` spell check integration on pull requests

### PR DESCRIPTION
**Pull request by goggle**
As suggested in https://github.com/JuliaDocs/Documenter.jl/pull/2088#issuecomment-1485964371, here is an attempt to set up a spell checking GitHub Action for pull requests.

Unfortunately, I'm not sure how to test if it works as intended...

Some comments on the exceptions defined in the config file `.typos.toml`:

- `fo` occurs in `test/utilities.jl`:
```
error: `fo` should be `of`, `for`
  --> ./test/utilities.jl:508:32
    |
508 |                 RemoteLibrary("fo\'o", "example.com\n/foo"),
    |                                ^^
    |
error: `fo` should be `of`, `for`
  --> ./test/utilities.jl:511:32
    |
511 |             push!(r, Snippet(["fo\'o"], ["Foo"], "f(x)"))
    |                                ^^
    |
error: `fo` should be `of`, `for`
  --> ./test/utilities.jl:516:30
    |
516 |             @test occursin("'fo\\'o'", output)
    |                              ^^
    |
error: `fo` should be `of`, `for`
  --> ./test/utilities.jl:518:31
    |
518 |             @test !occursin("'fo'o'", output)
    |                               ^^
    |
```

- `ket` occurs in `docs/src/man/latex.md`, `test/examples/make.jl` and `test/examples/src/man/tutorial.md`:
```
error: `ket` should be `kept`
  --> ./docs/src/man/latex.md:155:10
    |
155 |         :ket => ["|#1\\rangle", 1],
    |          ^^^
    |
error: `ket` should be `kept`
  --> ./test/examples/make.jl:260:22
    |
260 |                     :ket => ["|#1\\rangle", 1],
    |                      ^^^
    |
error: `ket` should be `kept`
  --> ./test/examples/make.jl:282:22
    |
282 |                     :ket => ["|#1\\rangle", 1],
    |                      ^^^
    |
error: `ket` should be `kept`
  --> ./test/examples/make.jl:309:26
    |
309 |                         :ket => ["|#1\\rangle", 1],
    |                          ^^^
    |
error: `ket` should be `kept`
  --> ./test/examples/src/man/tutorial.md:444:14
    |
444 |             :ket => ["|#1\\rangle", 1],
    |              ^^^
    |
```

- `trun` occurs in `test/utilities.jl` and `test/TestUtilities.jl`:
```
error: `trun` should be `turn`
  --> ./test/utilities.jl:171:19
    |
171 |             @test trun(`$(git()) init`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:172:19
    |
172 |             @test trun(`$(git()) config user.email "tester@example.com"`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:173:19
    |
173 |             @test trun(`$(git()) config user.name "Test Committer"`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:174:19
    |
174 |             @test trun(`$(git()) remote add origin git@github.com:JuliaDocs/Documenter.jl.git`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:178:19
    |
178 |             @test trun(`$(git()) add -A`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:179:19
    |
179 |             @test trun(`$(git()) commit -m"Initial commit."`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:201:19
    |
201 |             @test trun(`$(git()) worktree add $(path_worktree)`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:224:19
    |
224 |             @test trun(`$(git()) init`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:225:19
    |
225 |             @test trun(`$(git()) config user.email "tester@example.com"`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:226:19
    |
226 |             @test trun(`$(git()) config user.name "Test Committer"`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:234:19
    |
234 |             @test trun(`$(git()) -c protocol.file.allow=always submodule add $(path_repo) repository`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:235:19
    |
235 |             @test trun(`$(git()) add -A`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:236:19
    |
236 |             @test trun(`$(git()) commit -m"Initial commit."`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:265:19
    |
265 |             @test trun(`$(git()) init`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:266:19
    |
266 |             @test trun(`$(git()) config user.email "tester@example.com"`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:267:19
    |
267 |             @test trun(`$(git()) config user.name "Test Committer"`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:268:19
    |
268 |             @test trun(`$(git()) remote add origin git@this-is-not-github.com:JuliaDocs/Documenter.jl.git`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:272:19
    |
272 |             @test trun(`$(git()) add -A`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:273:19
    |
273 |             @test trun(`$(git()) commit -m"Initial commit."`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:580:19
    |
580 |             @test trun(`$(git()) -C $(path) init --bare`)
    |                   ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:591:23
    |
591 |                 @test trun(`$(git()) clone $(path) $(subdir_path)`)
    |                       ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:592:23
    |
592 |                 @test trun(`$(git()) -C $(subdir_path) config user.email "tester@example.com"`)
    |                       ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:593:23
    |
593 |                 @test trun(`$(git()) -C $(subdir_path) config user.name "Test Committer"`)
    |                       ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:594:23
    |
594 |                 @test trun(`$(git()) -C $(subdir_path) checkout -b $(head)`)
    |                       ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:595:23
    |
595 |                 @test trun(`$(git()) -C $(subdir_path) commit --allow-empty -m"initial empty commit"`)
    |                       ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:596:23
    |
596 |                 @test trun(`$(git()) -C $(subdir_path) push --set-upstream origin $(head)`)
    |                       ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:611:23
    |
611 |                 @test trun(`$(git()) clone barerepo/ local/`)
    |                       ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:614:23
    |
614 |                 @test trun(`$(git()) -C local/ remote add other ../barerepo_other/`)
    |                       ^^^^
    |
error: `trun` should be `turn`
  --> ./test/utilities.jl:615:23
    |
615 |                 @test trun(`$(git()) -C local/ fetch other`)
    |                       ^^^^
    |
error: `trun` should be `turn`
  --> ./test/TestUtilities.jl:5:18
  |
5 | export @quietly, trun
  |                  ^^^^
  |
error: `trun` should be `turn`
  --> ./test/TestUtilities.jl:115:10
    |
115 | function trun(cmd::Base.AbstractCmd)
    |          ^^^^
    |
``` 

- `varius` occurs in `docs/src/showcase.md`, `test/examples/src.latex_showcase/showcase.md` and `test/examples/references/latex_showcase.tex`:
```
error: `varius` should be `various`
  --> ./docs/src/showcase.md:135:12
    |
135 | * Morbi et varius nisl, eu semper orci.
    |            ^^^^^^
    |
error: `varius` should be `various`
  --> ./docs/src/showcase.md:143:52
    |
143 |   > Fusce tempus scelerisque egestas. Pellentesque varius nulla a varius fringilla.
    |                                                    ^^^^^^
    |
error: `varius` should be `various`
  --> ./docs/src/showcase.md:143:67
    |
143 |   > Fusce tempus scelerisque egestas. Pellentesque varius nulla a varius fringilla.
    |                                                                   ^^^^^^
    |
error: `varius` should be `various`
  --> ./docs/src/showcase.md:155:12
    |
155 | * Morbi et varius nisl, eu semper orci.
    |            ^^^^^^
    |
error: `varius` should be `various`
  --> ./docs/src/showcase.md:181:16
    |
181 |     * Morbi et varius nisl, eu semper orci.
    |                ^^^^^^
    |
error: `varius` should be `various`
  --> ./docs/src/showcase.md:197:14
    |
197 | > * Morbi et varius nisl, eu semper orci.
    |              ^^^^^^
    |
error: `varius` should be `various`
  --> ./test/examples/src.latex_showcase/showcase.md:134:12
    |
134 | * Morbi et varius nisl, eu semper orci.
    |            ^^^^^^
    |
error: `varius` should be `various`
  --> ./test/examples/src.latex_showcase/showcase.md:142:52
    |
142 |   > Fusce tempus scelerisque egestas. Pellentesque varius nulla a varius fringilla.
    |                                                    ^^^^^^
    |
error: `varius` should be `various`
  --> ./test/examples/src.latex_showcase/showcase.md:142:67
    |
142 |   > Fusce tempus scelerisque egestas. Pellentesque varius nulla a varius fringilla.
    |                                                                   ^^^^^^
    |
error: `varius` should be `various`
  --> ./test/examples/src.latex_showcase/showcase.md:154:12
    |
154 | * Morbi et varius nisl, eu semper orci.
    |            ^^^^^^
    |
error: `varius` should be `various`
  --> ./test/examples/src.latex_showcase/showcase.md:180:16
    |
180 |     * Morbi et varius nisl, eu semper orci.
    |                ^^^^^^
    |
error: `varius` should be `various`
  --> ./test/examples/src.latex_showcase/showcase.md:196:14
    |
196 | > * Morbi et varius nisl, eu semper orci.
    |              ^^^^^^
    |
error: `varius` should be `various`
  --> ./test/examples/references/latex_showcase.tex:324:16
    |
324 | \item Morbi et varius nisl, eu semper orci.
    |                ^^^^^^
    |
error: `varius` should be `various`
  --> ./test/examples/references/latex_showcase.tex:334:48
    |
334 | Fusce tempus scelerisque egestas. Pellentesque varius nulla a varius fringilla.
    |                                                ^^^^^^
    |
error: `varius` should be `various`
  --> ./test/examples/references/latex_showcase.tex:334:63
    |
334 | Fusce tempus scelerisque egestas. Pellentesque varius nulla a varius fringilla.
    |                                                               ^^^^^^
    |
error: `varius` should be `various`
  --> ./test/examples/references/latex_showcase.tex:363:16
    |
363 | \item Morbi et varius nisl, eu semper orci.
    |                ^^^^^^
    |
error: `varius` should be `various`
  --> ./test/examples/references/latex_showcase.tex:415:16
    |
415 | \item Morbi et varius nisl, eu semper orci.
    |                ^^^^^^
    |
error: `varius` should be `various`
  --> ./test/examples/references/latex_showcase.tex:450:16
    |
450 | \item Morbi et varius nisl, eu semper orci.
    |                ^^^^^^
    |
```

### CODE REVIEW

**Comment by goggle** [comment]
Ok, the action already got triggered (https://github.com/JuliaDocs/Documenter.jl/actions/runs/4543898985/jobs/8009202461?pr=2091), and it seems to work as intended.

**Comment by goggle** [comment]
After https://github.com/JuliaDocs/Documenter.jl/pull/2091/commits/073a3d8fa24a5e42a931c6e979dfcbf45a51618b the CI doesn't get triggered anymore, so I cannot confirm if it still works.

**Comment by mortenpi** [comment]
Could we have it in a separate workflow that only runs `on.pull_request`? It would be good if this didn't fail the CI on master.

**Comment by goggle** [comment]
Ok, I have reverted https://github.com/JuliaDocs/Documenter.jl/pull/2091/commits/073a3d8fa24a5e42a931c6e979dfcbf45a51618b

File: src/Builder.jl [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 
-Each stage of the pipeline performs an action on a [`Documenter.Document`](@ref) object.
+Each stge of the pipeline performs an action on a [`Documenter.Document`](@ref) object.
 These actions may involve creating directory structures, expanding templates, running
 doctests, etc.
 """
 module Builder
 
+# availble
+
```
Comment (mortenpi) [inline src/Builder.jl:11]:
```suggestion
Each stage of the pipeline performs an action on a [`Documenter.Document`](@ref) object.
These actions may involve creating directory structures, expanding templates, running
doctests, etc.
"""
module Builder

```

**Review by mortenpi — APPROVED** [review]
LGTM. Thanks @goggle!

**Pull request merged**